modules/ximgproc/test/test_rolling_guidance_filter.cpp

```diff
@@ -137,9 +137,9 @@ INSTANTIATE_TEST_CASE_P(TypicalSet1, RollingGuidanceFilterTest,
 //////////////////////////////////////////////////////////////////////////
 
 typedef tuple<double, string, int> RGFBFParam;
-typedef TestWithParam<RGFBFParam> RollingGuidanceFilterTest_BilateralRef;
+typedef TestWithParam<RGFBFParam> RollingGuidanceFilterTest_GaussianRef;
 
-TEST_P(RollingGuidanceFilterTest_BilateralRef, Accuracy)
+TEST_P(RollingGuidanceFilterTest_GaussianRef, Accuracy)
```

Comment:
I think there is a problem in previous test and problem is still here : test uses gaussainblur to test gaussianblur. May be add result data in opencv_extra is necessary (see https://github.com/opencv/opencv_contrib/blob/master/modules/ximgproc/test/test_joint_bilateral_filter.cpp#L170-L199)

Reply:
Thank you for pointing it out! I modified the test to use naive gaussian.
Please note, however, that I had to use [a large tolerance value of 0.02](https://github.com/opencv/opencv_contrib/pull/1745/commits/84265bd9d667885b24d718c653fecd6e6f2c01d4#diff-54b7f4ae0ded9b81e2867156cd335a39R16) because of low accuracy of cv::GaussianBlur() for uchar type. And I removed tests for `sigmaS==8.0` because it takes too much time (about 10 sec for CV_32FC3).

Reply:
Just and idea : You can run original code using this link with lena (img) as data  and save result (imgref). Don't you think that it will a better test compare opencv result with imgref?
@alalek

Reply:
IMHO, comparing with OpenCV gaussian blur is valid too (we checking that it is called on the first iteration)